sdk/tools/mkisofs/schilytools/libmdigest/byte_order.h

```diff
@@ -72,12 +72,12 @@ extern "C" {
 	defined(__POWERPC__) || defined(POWERPC) || defined(__powerpc) || \
 	defined(__powerpc__) || defined(__powerpc64__) || defined(__ppc__) || \
 	defined(__hpux) || defined(_MIPSEB) || defined(mc68000) || \
-	defined(__s390__) || defined(__s390x__) || defined(sel)
+	defined(__s390__) || defined(__s390x__) || defined(sel) || (defined(__arm64__) && defined(__APPLE__))
 #define	CPU_BIG_ENDIAN
 #define	IS_BIG_ENDIAN 1
 #define	IS_LITTLE_ENDIAN 0
```

Comment:
> Per my research all M1 processors are big endian.

It's actually Little Endian:

From https://developer.apple.com/documentation/apple-silicon/porting-your-macos-apps-to-apple-silicon
> Both Apple silicon and Intel-based Mac computers use the little-endian format for data, so you don’t need to make endian conversions in your code.

From https://www.quora.com/Is-Apples-new-M1-processor-big-endian-like-the-original-Motorola-microprocessor-or-little-endian-like-the-Intel-processor
> ARMv8 instruction set architecture, used in Apples CPUs, like all modern architectures is a little-endian architecture, just like x86, and unlike 68000 or ancient Apples.

Reply:
In that case I have no idea why Google told me they are big-endian when I asked it the question. How strange.